2214/2/battle_expert.rb

```diff
@@ -0,0 +1,48 @@
+require 'terminal-table'
+require_relative 'popular_words_counter'
+require_relative 'battler'
+require_relative 'battle'
+
+class BattleExpert
+  BATTLES_FOLDER = 'Battles'.freeze
+  def describe_battlers(top_bad_words)
+    puts Terminal::Table.new rows: sorted_battlers[0...top_bad_words].map { |battler| battler_as_row(battler) }
+  end
+
+  def find_popular_words(battler_name, top_words = 30)
+    if battlers_names.include? battler_name
+      PopularWordsCounter.new(battles, battler_name, top_words).count
+    else
+      puts "Я не знаю МЦ #{battler_name}. Зато мне известны:"
+      battlers_names.each { |battler| puts battler }
+    end
+  end
+
+  private
+
+  def sorted_battlers
+    battlers.sort_by(&:bad_words_per_round).reverse
+  end
+
+  def battlers
+    battlers_names.map { |battler_name| Battler.new(battler_name, battles) }
```

Comment:
I wouldn't add all the battles to each `Battler`. Instead, you need to add only battles that belong to this specific battler. This makes the code of `Battler` and dependent classes hard to read. They should care about filtering and similar stuff.

Let's rewrite this.